### PR TITLE
[DV/pwrmgr/sysrst_ctrl] resolved all comments in #12413

### PR DESCRIPTION
**Pull request by jeoongp**
Signed-off-by: Joshua Park <jeoong@google.com>
Since tons of merge conflicts were not easy to be resolved, I removed the existing branch and created a new one with the same name.
Existing comments : https://github.com/lowRISC/opentitan/pull/12413/

### CODE REVIEW

**Comment by jeoongp** [comment]
> > I am not clear of CI (Build Docker Containers) failure. Please let me know how to resolve it.
> 
> @jeoongp are you able to click on the "details" of the failure and see where that takes you? You might not have access yet.
> 
> In any case, CI fails once in awhile because it has a lot of external dependencies, so I've just retriggered it.

Build Docker Containers) was initially failed but somehow it has passed later automatically while running other checks. Now all checks have passed.

File: hw/top_earlgrey/dv/env/seq_lib/chip_sw_sysrst_ctrl_vseq.sv
Comment (jeoongp) [reply hw/top_earlgrey/dv/env/seq_lib/chip_sw_sysrst_ctrl_vseq.sv]: No trailing space is found in these 4 lines. I am not clear of what to resolve..

File: hw/top_earlgrey/dv/env/seq_lib/chip_sw_sysrst_ctrl_vseq.sv
Comment (jeoongp) [reply hw/top_earlgrey/dv/env/seq_lib/chip_sw_sysrst_ctrl_vseq.sv]: File has a newline at line 31. I am not clear of what to resolve..

File: sw/device/tests/meson.build
Comment (cindychip) [inline sw/device/tests/meson.build:1166]: Can you also add a bazel BUILD rules?
SW team is transitioning to bazel tools.
Example: https://github.com/lowRISC/opentitan/blob/master/sw/device/tests/sim_dv/BUILD

File: hw/top_earlgrey/dv/env/seq_lib/chip_sw_sysrst_ctrl_vseq.sv
Comment (cindychip) [inline hw/top_earlgrey/dv/env/seq_lib/chip_sw_sysrst_ctrl_vseq.sv:26]: Could you remove this if not needed?

**Comment by tjaychen** [comment]
@jeoongp are you missing one of the sequence files? it looks like the `all_reset` one might not be here?

File: hw/top_earlgrey/dv/env/seq_lib/chip_sw_sysrst_ctrl_vseq.sv
Comment (jeoongp) [reply hw/top_earlgrey/dv/env/seq_lib/chip_sw_sysrst_ctrl_vseq.sv]: @cindychip a good catch!. I fixed it.

File: sw/device/tests/meson.build
Comment (jeoongp) [reply sw/device/tests/meson.build:1166]: Yes, I added the bazel BUILD rules. Thanks

Comment (jeoongp) [reply sw/device/tests/meson.build:1166]: It's passing

File: hw/top_earlgrey/dv/env/ast_supply_if.sv
Comment (tjaychen) [inline hw/top_earlgrey/dv/env/ast_supply_if.sv:76]: do we still need the internal force now? it looks like there's a pin force below already.

File: hw/top_earlgrey/dv/env/seq_lib/chip_sw_sysrst_ctrl_vseq.sv
Comment (tjaychen) [inline hw/top_earlgrey/dv/env/seq_lib/chip_sw_sysrst_ctrl_vseq.sv:24]: is this 1000us related to the internal debounce? 
if yes, maybe we can talk to @msfschaffner about whether there's a better mechanism to handle the delay, since this may change based on the clock. 

This can probably be done as a separate PR.

File: hw/top_earlgrey/dv/env/ast_supply_if.sv
Comment (jeoongp) [reply hw/top_earlgrey/dv/env/ast_supply_if.sv:76]: We don't need this code. I fixed it.

File: sw/device/tests/pwrmgr_all_reset_reqs_test.c
Comment (tjaychen) [inline sw/device/tests/pwrmgr_all_reset_reqs_test.c:1]: @jeoongp i'm a little bit confused by this test, how is this one different from `aon_timer_wdog_bite_reset_test.c`?

File: hw/top_earlgrey/dv/env/seq_lib/chip_sw_sysrst_ctrl_vseq.sv
Comment (jeoongp) [reply hw/top_earlgrey/dv/env/seq_lib/chip_sw_sysrst_ctrl_vseq.sv:24]: 1000 us is a time delay between internal SW configuration and external key input. 
Does the internal debounce mean the key pressed time? If yes, it's handled by
line 26 : cfg.ast_supply_vif.pulse_key0_i_next_trigger(50 + cycles_after_trigger);  

Yes, we can use the better mechanism for both delays.

File: sw/device/tests/pwrmgr_all_reset_reqs_test.c
Comment (jeoongp) [reply sw/device/tests/pwrmgr_all_reset_reqs_test.c:1]: This test does not include deep sleep waked up by watch dog. It tests the reset by watch dog timer.

Comment (tjaychen) [reply sw/device/tests/pwrmgr_all_reset_reqs_test.c:1]: o got it. Should the test name be a bit different? This test name seems to imply all reset requests, but it looks like it's more focused on specifically on the wdog.

**Review by tjaychen — APPROVED** [review]
lgtm. 
i have a couple of clarifying questions.

I think most of my comments can be addressed in another PR, it doesn't have to block this one from merging.

File: hw/top_earlgrey/dv/env/ast_supply_if.sv
Comment (tjaychen) [reply hw/top_earlgrey/dv/env/ast_supply_if.sv:76]: @jeoongp looks like the task was removed but the invocation is still here :)

Comment (jeoongp) [reply hw/top_earlgrey/dv/env/ast_supply_if.sv]: Just fixed it. Thanks!

File: hw/top_earlgrey/dv/env/seq_lib/chip_sw_sysrst_ctrl_vseq.sv
Comment (jeoongp) [reply hw/top_earlgrey/dv/env/seq_lib/chip_sw_sysrst_ctrl_vseq.sv:24]: I see. If we need to use another approach on these fixed delays in another PR, let me know.
@msfschaffner

Comment (msfschaffner) [reply hw/top_earlgrey/dv/env/seq_lib/chip_sw_sysrst_ctrl_vseq.sv:24]: It looks like this is an arbitrary delay between SW configuration and stimulus application on key0, right? Does it maybe make sense to tie this to AON clock cycles and randomize the amount of clock cycles that we wait?

File: sw/device/tests/sim_dv/pwrmgr_sysrst_ctrl_test.c
Comment (jeoongp) [reply sw/device/tests/sim_dv/pwrmgr_sysrst_ctrl_test.c:158]: I did not know the light sleep but I agree that we need to check all these modes. I will cover it in new tests apart from this PR.

**Comment by tjaychen** [comment]
should we merge this if there are no objections?

File: sw/device/tests/sim_dv/pwrmgr_deep_sleep_sysrst_reqs_test.c
Comment (jdonjdon) [reply sw/device/tests/sim_dv/pwrmgr_deep_sleep_sysrst_reqs_test.c:183]: Will check with a separate test.
It is in spread sheet but haven't implement yet.

File: sw/device/tests/sim_dv/pwrmgr_sysrst_ctrl_test.c
Comment (jdonjdon) [reply sw/device/tests/sim_dv/pwrmgr_sysrst_ctrl_test.c:158]: > 

yes, all deep sleep test have their pair of nomal sleep test in the spread sheet.
AFAIK

Comment (tjaychen) [reply sw/device/tests/sim_dv/pwrmgr_sysrst_ctrl_test.c:158]: i think one thing that's worth discussing..and this is a matter of personal philosophy.
I generally prefer a single test to test all levels of sleep. This ensures that we don't have bridging / transition issues.

My worry is that if they're all separate tests, we always start from a clean slate, and that might mask problems. (i'm very suspicious of pwrmgr/rstmgr in this area).

**Review by cindychip — APPROVED** [review]

**Comment by tjaychen** [comment]
merging! thanks for all the changes @jeoongp

File: sw/device/tests/sim_dv/pwrmgr_sysrst_ctrl_test.c
Comment (jeoongp) [reply sw/device/tests/sim_dv/pwrmgr_sysrst_ctrl_test.c:158]: @tjaychen I agree. I will add a single test to check all sleeps. Thanks!

**Comment by jeoongp** [comment]
Thanks @tjaychen @jdonjdon for reviews/help!

File: hw/top_earlgrey/dv/env/seq_lib/chip_sw_sysrst_ctrl_vseq.sv
Comment (msfschaffner) [reply hw/top_earlgrey/dv/env/seq_lib/chip_sw_sysrst_ctrl_vseq.sv:24]: As for the fixed part, we need to make sure that the config is actually visible on the AON side.
This is basically the CDC delay, which is in the order of ~3 AON clock cycles.
Hence, I would suggest to make the fixed part 10 AON cycles.
The variable part depends a bit how long the sims take.
We can go up to 1ms (200 cycles) as you suggested, and make it shorter if it takes too long...

File: sw/device/tests/BUILD
Comment (timothytrippel) [reply sw/device/tests/BUILD:146]: Sorry just saw this. To specify a test that only works for DV (which this one is such a test), we would want to:
1. add the `targets = ["dv"],` line to this rule, and
2. move this entire test to `sw/device/tests/sim_dv` to keep it with the other tests that only run in DV

Comment (timothytrippel) [reply sw/device/tests/BUILD:146]: @tjaychen is correct, we shouldn't say `failing_verilator` here, nor should be pass any special verilator params since this test only runs in DV sim

File: sw/device/tests/sim_dv/pwrmgr_deep_sleep_sysrst_reqs_test.c
Comment (jeoongp) [reply sw/device/tests/sim_dv/pwrmgr_deep_sleep_sysrst_reqs_test.c:183]: Normal sleep and deep sleep are implemented in https://github.com/lowRISC/opentitan/pull/12757

File: hw/top_earlgrey/dv/env/seq_lib/chip_sw_sysrst_ctrl_vseq.sv
Comment (jeoongp) [reply hw/top_earlgrey/dv/env/seq_lib/chip_sw_sysrst_ctrl_vseq.sv:24]: Thanks @msfschaffner. Your suggestion is implemented in https://github.com/lowRISC/opentitan/pull/12757.

File: sw/device/tests/sim_dv/pwrmgr_sysrst_ctrl_test.c
Comment (jeoongp) [reply sw/device/tests/sim_dv/pwrmgr_sysrst_ctrl_test.c:158]: This single test is implemented in https://github.com/lowRISC/opentitan/pull/12757

**Pull request merged**
